src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                        >
+                            LOGO
+                        </Typography>
+                        
+                        <SmallView>
+                        <Box>                                
+                            {toggleButton}
+                        </Box>
+                        <IconButton
+                            sx={style.margin.iconBtn}
+                            aria-label="open drawer"
+                            edge="start"
+                            onClick={handleDrawerToggle}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        </SmallView>
+                        <Responsive>
+                            <Box>
+                                <Button>
+                                    <Link style={Links} to="/posts/general">
+                                        General
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/learn">
+                                        Learn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/interview">
+                                        Interview
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/project">
+                                        Project
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/login">
+                                        SignIn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/register">
+                                        SignUp
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/my-account/1">
+                                        Profile
+                                    </Link>
+                                </Button>
+                                {toggleButton}
+                            </Box>
+                        </Responsive>
+                    </Toolbar>
+                </Container>
+            </AppBar>
+            <SmallView>
+            <Box component="nav">
+                <Drawer
+                    anchor={"right"}
+                    container={container}
```

Comment:
It shouldn't be necessary to specify container as window.document.body as this is the default for container:
<img width="815" alt="image" src="https://user-images.githubusercontent.com/25972701/181915948-9fc1518e-50c0-4ad4-a2d1-43d242d9ed98.png">
https://mui.com/material-ui/api/modal/
Drawer implements a Modal so it also uses the container prop from modal.